cfme/containers/provider/__init__.py

```diff
@@ -355,6 +361,44 @@ def step(self):
         sel.click(InfoBlock('Overview', 'Topology'))
 
 
+class AdHocMetricsView(BaseLoggedInPage):
+    filter_dropdown = SelectorDropdown('uib-tooltip', 'Filter by')
+    apply_btn = form_buttons.apply_filters
+    selected_filter = None
+
+    @property
+    def is_displayed(self):
+        return False
+
+    def wait_for_filter_option_to_load(self):
+        wait_for(lambda: bool(self.filter_dropdown.items), delay=5, num_sec=60)
+
+    def wait_for_results_to_load(self):
+        wait_for(lambda: bool(int(Text(self, 'h5.ng-binding').text.split()[0])),
+                 delay=5, num_sec=60)
+
+    def apply_filter(self):
+        form_buttons._fill_fb_bool(self.apply_btn, True)
+
+    def set_random_filter(self):
+
+        random_filter = str(random.choice(self.filter_dropdown.items))
+        self.selected_filter = random_filter
+        self.filter_dropdown.fill_with(random_filter)
```

Comment:
Something like this?
this will be the view's function:
`def set_filter(self, filter):`
`def get_random_filter(self):`

and the call from the test will be like:
`view.set_filter(view.get_random_filter)`